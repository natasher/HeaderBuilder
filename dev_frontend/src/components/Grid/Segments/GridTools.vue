<template>
<div class = "tools">

  <a href = "#"
    v-if  = "getGridStatus === 'custom'"
    @click.passive = "showGridOptionsModal({ $modal: $modal })" >
      <font-icon icon = "pencil" />
  </a>

</div>
</template>

<script>
import { mapActions, mapGetters } from 'vuex';
import FontIcon from '../../BaseGeneric/FontIcon.vue';

export default {

  name: 'GridTools',

  components: {
    FontIcon,
  },

  computed: {
    ...mapGetters( 'ui', [ 'getGridStatus' ]),
  },

  methods: {
    ...mapActions( 'modals', [ 'showGridOptionsModal' ]),
  },

}
</script>
