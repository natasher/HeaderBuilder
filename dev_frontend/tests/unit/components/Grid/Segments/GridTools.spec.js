import { shallowMount, createLocalVue } from '@vue/test-utils'
import merge         from 'lodash.merge'
import flushPromises from 'flush-promises'
import Vuex          from 'vuex'
import GridTools     from '../../../../../src/components/Grid/Segments/GridTools.vue'

const localVue = createLocalVue()
localVue.use( Vuex )

const $modal = jest.fn()

function createStore(overrides) {
  const defaultStoreConfig = {
    modules: {
      ui: {
        namespaced: true,
        getters: {
          getGridStatus: jest.fn(() => { return 'custom' }),
        }
      },
      modals: {
        namespaced: true,
        actions: {
          showGridOptionsModal: jest.fn(),
        }
      },
    }
  }

  return new Vuex.Store( merge( defaultStoreConfig, overrides ))
}

function createWrapper(overrides) {
  const defaultMountingOptions = {
    localVue,
    store: createStore(),
    mocks: {
      $modal,
    }
  }

  return shallowMount( GridTools, merge( defaultMountingOptions, overrides ))
}

describe('GridTools.vue', () => {

  test('should match snapshot if grid status is `custom`', () => {
    const wrapper = createWrapper()

    expect( wrapper.element ).toMatchSnapshot()
  })

  test('should match snapshot if grid status is NOT `custom`', () => {
    const store = createStore({
      modules: {
        ui: {
          getters: {
            getGridStatus: jest.fn(() => { return 'test' })
          }
        }
      }
    })
    const wrapper = createWrapper({ store, })

    expect( wrapper.element ).toMatchSnapshot()
  })

  test('should open grid options modal if edit icon is clicked', async () => {
    const showGridOptionsModal = jest.fn()
    const store = createStore({
      modules: {
        modals: {
          actions: {
            showGridOptionsModal,
          }
        }
      }
    })
    const wrapper = createWrapper({ store, })
    const btn = wrapper.find('a')

    btn.trigger( 'click' )
    await flushPromises()

    expect(showGridOptionsModal).toBeCalledWith(
      expect.anything(),
      { $modal, },
      void 0
    )
  })

})