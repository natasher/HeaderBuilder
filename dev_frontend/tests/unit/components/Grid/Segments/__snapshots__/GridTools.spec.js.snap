// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridTools.vue should match snapshot if grid status is \`custom\` 1`] = `
<div
  class="tools"
>
  <a
    href="#"
  >
    <fonticon-stub
      icon="pencil"
    />
  </a>
</div>
`;

exports[`GridTools.vue should match snapshot if grid status is NOT \`custom\` 1`] = `
<div
  class="tools"
>
  <!---->
</div>
`;
